From 697d1819f97058745eddabc44842529f6a239ba9 Mon Sep 17 00:00:00 2001
From: Biswapriyo Nath <nathbappai@gmail.com>
Date: Mon, 30 Jan 2023 14:20:50 +0530
Subject: [PATCH] meson: Fix linking flag in pkgconfig for mingw

This changes the extra libs in pkgconfig from -Wl,-luuid to -luuid.
Because the uuid is a system library and not a linker flag. This
also fixes the following error while using gtk3 in golang program.
go build github.com/gotk3/gotk3/gdk: invalid flag in pkg-config --libs: -Wl,-luuid
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6c711e9e9a3..d703e377843 100644
--- a/meson.build
+++ b/meson.build
@@ -664,7 +664,7 @@ if win32_enabled
     # link on Visual Studio builds
     extra_demo_ldflags = ['/entry:mainCRTStartup']
   else
-    pc_gdk_extra_libs += ['-Wl,-luuid']
+    pc_gdk_extra_libs += ['-luuid']
   endif
   pc_gdk_extra_libs += ['-lwinmm', '-ldwmapi', '-lsetupapi', '-lcfgmgr32', '-lhid', '-lwinspool', '-lcomctl32', '-lcomdlg32']
   backend_immodules += ['ime']
-- 
GitLab

